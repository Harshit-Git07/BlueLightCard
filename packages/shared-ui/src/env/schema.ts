import { z } from 'zod';
import { EnvVarObject } from './types';

// shared env vars
export const sharedEnvVars: EnvVarObject = {
  APP_BRAND: process.env.NEXT_PUBLIC_APP_BRAND || process.env.STORYBOOK_APP_BRAND,
};

export const envSchema = z.object({
  APP_BRAND: z.enum(['blc-uk-current', 'blc-uk', 'blc-au', 'dds-uk']).default('blc-uk'),
});
