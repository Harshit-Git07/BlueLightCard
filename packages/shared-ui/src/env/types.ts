import { ZodTypeAny } from 'zod';

export type EnvVarObject = Record<string, string | undefined>;
export type ExtendEnvVars<K extends string, T extends ZodTypeAny> = Record<
  K,
  {
    value: string | undefined;
    schema: T;
  }
>;
