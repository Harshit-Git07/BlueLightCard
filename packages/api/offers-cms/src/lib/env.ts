import { createEnv } from '@t3-oss/env-core';
import { z } from 'zod';

export const env = createEnv({
  server: {
    BRAND: z.string(),
    OFFERS_DISCOVERY_EVENT_BUS_NAME: z.string().optional(),
    OFFERS_CMS_ACCOUNT: z.string().optional(),
    DWH_FIREHOSE_COMPANY_LOCATION_STREAM_NAME: z.string().optional(),
    DWH_FIREHOSE_MENU_STREAM_NAME: z.string().optional(),
    DWH_FIREHOSE_THEMED_MENU_STREAM_NAME: z.string().optional(),
  },

  runtimeEnv: process.env,
  emptyStringAsUndefined: true,
});
