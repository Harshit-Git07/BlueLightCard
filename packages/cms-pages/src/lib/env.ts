export default process.env.NODE_ENV === 'development' || process.env.ENABLE_PREVIEW === 'true';

export const APP_BRAND = process.env.NEXT_PUBLIC_APP_BRAND ?? 'blc-uk';
