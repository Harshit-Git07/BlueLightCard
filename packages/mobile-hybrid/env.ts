import { BrandToken } from '@bluelightcard/shared-ui/tailwind';
import { z } from 'zod';

export const env = z
  .object({
    NEXT_APP_BRAND: z.enum([BrandToken.BLC, BrandToken.DDS]).optional(),
  })
  .parse(process.env);

export const BRAND = env.NEXT_APP_BRAND ?? BrandToken.BLC;
