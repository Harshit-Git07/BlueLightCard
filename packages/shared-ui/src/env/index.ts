import { envSchema, sharedEnvVars } from './schema';

export const env = envSchema.parse(sharedEnvVars);

export * from './utils';
