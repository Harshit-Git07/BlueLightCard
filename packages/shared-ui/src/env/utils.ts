import { merge } from 'lodash';
import { ZodTypeAny, z } from 'zod';
import { ExtendEnvVars, EnvVarObject } from './types';
import { envSchema, sharedEnvVars } from './schema';

/**
 * Merges shared env vars per consumption
 * @param extendEnvVars
 */
export function envMerge<K extends string, T extends ZodTypeAny>(
  mergableEnvVars: ExtendEnvVars<K, T>,
) {
  const zodSchema = {} as Record<K, T>;
  const envVars: EnvVarObject = {};

  for (const envVarName in mergableEnvVars) {
    zodSchema[envVarName] = mergableEnvVars[envVarName].schema;
    envVars[envVarName] = mergableEnvVars[envVarName].value;
  }

  const extendedSchema = envSchema.merge(z.object(zodSchema));

  return extendedSchema.parse(merge(sharedEnvVars, envVars));
}
