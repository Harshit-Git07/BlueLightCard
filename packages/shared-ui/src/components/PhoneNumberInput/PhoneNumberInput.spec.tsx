import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import PhoneNumberInputDataManager, { getLocalCountryCode } from './';
import { env } from '../../env';

describe('PhoneNumberInput', () => {
  const defaultProps = {
    id: 'my-phone-number',
    disabled: false,
    showErrors: false,
    defaultCountry: 'us',
    emptyErrorMessage: 'Please enter a phone number',
    invalidErrorMessage: 'Please enter a valid phone number',
    label: 'Phone Number',
    helpText: 'Enter your phone number',
    helpIcon: true,
    messageText: '',
    phoneNumber: '',
    dialCode: '+1',
    onPhoneChange: jest.fn(),
    isSelectable: true,
  };

  it('should render correctly', () => {
    render(<PhoneNumberInputDataManager {...defaultProps} />);
    const label = screen.getByLabelText(/Phone Number/i, {});
    expect(label).toBeInTheDocument();
  });

  it('should render input fields with initial values', () => {
    render(<PhoneNumberInputDataManager {...defaultProps} />);
    const dialCodeInput = screen.getByPlaceholderText('+XX', {});
    const phoneNumberInput = screen.getByPlaceholderText('XXXXXXXXXX', {});

    expect(dialCodeInput).toBeInTheDocument();
    expect(phoneNumberInput).toBeInTheDocument();
    expect(dialCodeInput).toHaveValue('+1');
  });

  it('should handle phone number change', () => {
    render(<PhoneNumberInputDataManager {...defaultProps} />);
    const phoneNumberInput = screen.getByPlaceholderText('XXXXXXXXXX', {});

    fireEvent.change(phoneNumberInput, { target: { value: '1234567890' } });
    expect(phoneNumberInput).toHaveValue('(123) 456-7890');
  });

  it('should show error when phone number is invalid', () => {
    render(<PhoneNumberInputDataManager {...defaultProps} showErrors={true} />);
    const phoneNumberInput = screen.getByPlaceholderText('XXXXXXXXXX', {});

    fireEvent.change(phoneNumberInput, { target: { value: '123' } });
    fireEvent.blur(phoneNumberInput);

    expect(screen.getByText(/Please enter a valid phone number/i, {})).toBeInTheDocument();
  });

  it('should show empty error message when no phone number is entered', () => {
    render(<PhoneNumberInputDataManager {...defaultProps} showErrors={true} />);
    const phoneNumberInput = screen.getByPlaceholderText('XXXXXXXXXX', {});

    fireEvent.change(phoneNumberInput, { target: { value: '' } });
    fireEvent.blur(phoneNumberInput);

    expect(screen.getByText(/Please enter a phone number/i, {})).toBeInTheDocument();
  });

  it('should handle dial code change', () => {
    render(<PhoneNumberInputDataManager {...defaultProps} />);
    const dialCodeInput = screen.getByPlaceholderText('+XX', {});

    fireEvent.change(dialCodeInput, { target: { value: '+44' } });
    expect(dialCodeInput).toHaveValue('+44');
  });

  it('should not allow interaction when disabled', () => {
    render(<PhoneNumberInputDataManager {...defaultProps} disabled={true} />);
    const phoneNumberInput = screen.getByPlaceholderText('XXXXXXXXXX', {});
    expect(phoneNumberInput).toBeDisabled();
  });

  it('should open and close country dropdown on flag click', () => {
    render(<PhoneNumberInputDataManager {...defaultProps} />);

    const countryButton = screen.getByRole('button', { name: /Country selector/i });

    fireEvent.click(countryButton);

    expect(screen.getByLabelText(/Country selector/i, {})).toBeInTheDocument();
  });

  it('should render country dropdown when isSelectable is true', () => {
    render(<PhoneNumberInputDataManager {...defaultProps} isSelectable={true} />);
    const countryButton = screen.getByRole('button', { name: /Country selector/i });
    fireEvent.click(countryButton);
    expect(countryButton).toBeInTheDocument();
    expect(countryButton).toHaveClass('flex flex-row items-center cursor-pointer false');
  });

  it('should not render country dropdown when isSelectable is false', () => {
    render(<PhoneNumberInputDataManager {...defaultProps} isSelectable={false} />);

    const countryButton = screen.queryByRole('button', { name: /Country selector/i });

    expect(countryButton).toHaveClass('flex flex-row items-center cursor-default false');
  });

  describe('returns related to country', () => {
    beforeEach(() => {
      jest.clearAllMocks();
    });

    it('returns "gb" for APP_BRAND "blc-uk"', () => {
      env.APP_BRAND = 'blc-uk';
      Object.defineProperty(navigator, 'language', {
        value: 'en-GB',
        configurable: true,
      });

      expect(getLocalCountryCode()).toBe('gb');
    });

    it('returns "au" for APP_BRAND "blc-au"', () => {
      env.APP_BRAND = 'blc-au';
      Object.defineProperty(navigator, 'language', {
        value: 'en-AU',
        configurable: true,
      });

      expect(getLocalCountryCode()).toBe('au');
    });
  });
});
