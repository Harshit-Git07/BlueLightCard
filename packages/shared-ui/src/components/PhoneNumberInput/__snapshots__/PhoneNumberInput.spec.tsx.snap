// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneNumberInput matches snapshots 1`] = `
<div>
  <div
    class="max-w-[524px] flex gap-2 flex-col"
  >
    <div
      class="flex flex-col gap-2"
    >
      <div
        class="flex items-center gap-[7px]"
      >
        <label
          aria-label="Phone Number"
          class="text-typography-body
   font-typography-body
   font-typography-body-weight
   leading-text-typography-body
   tracking-text-typography-body text-colour-onSurface dark:text-colour-onSurface-dark"
        >
          Phone Number
        </label>
        <div
          class="group relative inline-block"
        >
          <div
            aria-describedby=":r0:"
          >
            <svg
              aria-hidden="true"
              aria-label="information"
              class="svg-inline--fa fa-circle-info h-3.5 text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
              data-icon="circle-info"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="w-max left-full top-1/2 z-50 ml-3 -translate-y-1/2 transition-opacity ease-in pointer-events-none max-w-60 text-center rounded px-4 py-[6px] opacity-0 group-hover:opacity-100 absolute bg-colour-surface-inverse dark:bg-colour-surface-inverse-dark dark:text-colour-onSurface text-colour-onSurface-dark font-typography-body-small font-typography-body-small-weight text-typography-body-small leading-typography-body-small"
            id=":r0:"
            role="tooltip"
          >
            <span
              class="bg-colour-surface-inverse dark:bg-colour-surface-inverse-dark rotate-45 rounded-sm absolute left-[-3px] top-1/2 -z-10 h-2 w-2 -translate-y-1/2"
            />
            Enter your phone number
          </div>
        </div>
      </div>
      <div
        class="relative"
      >
        <span
          class="flex items-center gap-2 px-5 border rounded bg-colour-surface-brand-light dark:bg-colour-surface-brand-dark 
          border-onSurface-outline-light dark:border-onSurface-outline-dark 
        false 
        false"
        >
          <button
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Country selector"
            class="flex flex-row items-center cursor-pointer false "
            data-country="us"
            title="United States"
            type="button"
          >
            <img
              alt=""
              class="react-international-phone-flag-emoji h-full w-8 rounded-sm"
              data-country="us"
              draggable="false"
              loading="lazy"
              src="https://cdnjs.cloudflare.com/ajax/libs/twemoji/14.0.2/svg/1f1fa-1f1f8.svg"
            />
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-caret-down fa-2xs h-3 w-3 text-colour-onSurface-light dark:text-colour-onSurface-dark p-1"
              data-icon="caret-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                fill="currentColor"
              />
            </svg>
          </button>
          <input
            class="outline-none rounded ml-2 
          text-typography-body-large font-typography-body-large-weight leading-typography-body-large font-typography-body-large 
          bg-colour-surface-light dark:bg-colour-surface-dark 
           text-colour-surface-dark dark:text-colour-surface-light"
            placeholder="+XX"
            style="width: 25px;"
            type="tel"
            value="+1"
          />
          <input
            class="outline-none w-full py-3 mr-2 rounded 
            false 
            text-typography-body-large font-typography-body-large-weight leading-typography-body-large font-typography-body-large
           text-colour-surface-dark dark:text-colour-surface-light
            bg-colour-surface-brand-light dark:bg-colour-surface-brand-dark"
            placeholder="XXXXXXXXXX"
            type="tel"
            value=""
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
